parlai/core/teachers.py

```diff
@@ -256,7 +262,7 @@ def __init__(self, opt, shared=None):
         else:
             self.index = AttrDict(value=-1)
 
-        if not hasattr(self, 'data_loader'):
+        if not hasattr(self, 'data_loader') and opt['numworkers'] <= 1:
```

Comment:
we cannot mix QueueWorld with the dataloader in its current form, as it cannot be pickled.

Reply:
Cute. 😡

Reply:
Why can’t it be pickled?

Reply:
The Thread RLock's cannot be pickled specifically (i.e. the normal queue.Queue is not picklable)